src/ansi-c/expr2c.cpp

```diff
@@ -378,7 +383,10 @@ std::string expr2ct::convert_rec(
 
     dest+=" }";
 
-    dest+=d;
+    if(src.find(ID_typedef).is_not_nil())
+      dest+=src.get_string(ID_typedef);
+    else
+      dest+=d;
```

Comment:
This does appear to be a bug, for the following code:

    typedef struct tag_struct_name
    {
    	int x;
    	float y;
    } MYSTRUCT;
    
    void fun()
    {
    	   const struct tag_struct_name tag_struct_var = {.x = 1, .y = 3.14f};
    	   const MYSTRUCT mystruct_var = {.x = 3, .y = 2.1f};
    }

I get the following symbols:

    Symbol......: fun::1::mystruct_var
    Pretty name.: fun::1::mystruct_var
    Module......: main
    Base name...: mystruct_var
    Mode........: C
    Type........: const MYSTRUCT
    Value.......: { .x=3, .y=2.100000e+0f }
    Flags.......: lvalue thread_local file_local
    Location....: file main.c line 11 function fun
    
    Symbol......: fun::1::tag_struct_var
    Pretty name.: fun::1::tag_struct_var
    Module......: main
    Base name...: tag_struct_var
    Mode........: C
    Type........: struct tag_struct_name
    Value.......: { .x=1, .y=3.140000e+0f }
    Flags.......: lvalue thread_local file_local
    Location....: file main.c line 10 function fun